File: src/Application Specific/Classes/Entity Collections/Notes/NotesValidator.vb
Namespace Entity

    Namespace Notes

        Public Class NotesValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oNotes As Notes)

                'make sure that contact object is valid
                If oNotes.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oNotes.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If Not oNotes.CategoryID > 0 Then
                    Me.AddCriticalMessage("* Category Missing")
                End If
                If oNotes.Note.Trim.Length = 0 Then
                    Me.AddCriticalMessage("* Note Missing")
                End If
                If oNotes.UserIDFor = 0 Then
                    Me.AddCriticalMessage("* User For Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Notes/NotesValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Notes
    {
        // make sure that contact object is valid
        public class NotesValidator : BaseValidator
        {
            public void Validate(Notes oNotes)
            {
                // make sure that contact object is valid
                if (oNotes.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oNotes.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (!(oNotes.CategoryID > 0))
                {
                    AddCriticalMessage("* Category Missing");
                }

                if (oNotes.Note.Trim().Length == 0)
                {
                    AddCriticalMessage("* Note Missing");
                }

                if (oNotes.UserIDFor == 0)
                {
                    AddCriticalMessage("* User For Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}